lib/dor/models/editable.rb

```diff
@@ -232,16 +232,51 @@ def use_license=(use_license_machine)
     # @return [String] A description of the rights defined in the default object rights datastream. Can be 'Stanford', 'World', 'Dark' or 'None'
     def default_rights
       xml = defaultObjectRights.ng_xml
-      if xml.search('//rightsMetadata/access[@type=\'read\']/machine/group').length == 1
-        'Stanford'
-      elsif xml.search('//rightsMetadata/access[@type=\'read\']/machine/world').length == 1
-        'World'
-      elsif xml.search('//rightsMetadata/access[@type=\'discover\']/machine/none').length == 1
-        'Dark'
+      machine_read_access = xml.search('//rightsMetadata/access[@type="read"]/machine')
+      machine_discover_access = xml.search('//rightsMetadata/access[@type="discover"]/machine')
+
+      machine_read_access_node = machine_read_access.length == 1 ? machine_read_access.first : nil
+      machine_discover_access_node = machine_discover_access.length == 1 ? machine_discover_access.first : nil
+
+      if machine_read_access_node && machine_read_access_node.search('./group[text()="Stanford" or text()="stanford"]').length == 1
+        if machine_read_access_node.search('./group[@rule="no-download"]').length == 1
+          'stanford-nd'
+        else
+          'stanford'
+        end
+      elsif machine_read_access_node && machine_read_access_node.search('./world').length == 1
+        if machine_read_access_node.search('./world[@rule="no-download"]').length == 1
+          'world-nd'
+        else
+          'world'
+        end
+      elsif machine_read_access_node && machine_read_access_node.search('./location[text()="spec"]').length == 1
+        'loc:spec'
+      elsif machine_read_access_node && machine_read_access_node.search('./location[text()="music"]').length == 1
+        'loc:music'
+      elsif machine_discover_access_node && machine_discover_access_node.search('./world').length == 1
+        'none'
+      elsif machine_discover_access_node && machine_discover_access_node.search('./none').length == 1
+        'dark'
```

Comment:
i was just trying to cut down on the full xpath searches and samey xpath statements (particularly, what would've been some long and similar xpath queries for the `stanford` rights types).  also wanted to make more explicit that the conditions are working within either the read access or discover access nodes.  and since it's possible that either the read access node or the discover access node might be `nil`, i was guarding against referencing a method on a `nil` object with the `machine_read_access_node &&` and `machine_discover_access_node &&` parts of the conditions.